update quickstart to reflect new onboarding flow (#1274)
semgrep/semgrep-docs: docs/getting-started/quickstart.md

---
slug: quickstart
append_help_link: true

> title: Scan your first project with Semgrep

Author: I'm open to other titles for this page

Reviewer: Hola. I've encountered repo vs project cases before.

In Semgrep Cloud Platform, repositories are a superset of all the repos that a user can onboard (from their source code manager), whereas a Project is specifically "a repo that has been added to Semgrep".

"Scan your first repository with Semgrep" maybe?

Reviewer: We could also just call it ["Quickstart"](https://docs.snyk.io/getting-started/quickstart)

hide_title: false

> description: Learn how to set up Semgrep and scan your first project

Reviewer: ```suggestion
description: Learn how to set up Semgrep and scan your first project.
```
(To update if we change the term "project".)

Author: Changed this to repository based on your suggestion above.

tags:
  - quickstart
---

import MoreHelp from "/src/components/MoreHelp"
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

Learn how to set up Semgrep, scan your first project for security issues, and view your findings.

:::info prerequisites

> You must have Python 3.8 or later installed on the machine where the Semgrep CI is running.

Reviewer: ```suggestion
You must have Python 3.8 or later installed on the machine where the Semgrep CI is running.
```
What's actually performing the scan here - the CLI or the CI?

> You must have Python 3.7 or later installed on the machine where the Semgrep CI is running.

Reviewer: ```suggestion
You must have Python 3.8 or later installed on the machine where the Semgrep CI is running.
```

:::

> 1. Navigate to [Semgrep](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to grant Semgrep the needed permissions and proceed.

Reviewer: ```suggestion
1. Navigate to [Semgrep Cloud Platform (SCP)](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to grant Semgrep the needed permissions and proceed.
```

Alternatively: If the wordiness of Cloud Platform bothers you, we can also say "Sign into your Semgrep account".

> 1. Navigate to [Semgrep](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to grant Semgrep the needed permissions and proceed.

Reviewer: ```suggestion
1. Navigate to [Semgrep](https://semgrep.dev/login), and sign up by clicking on **Sign in with GitHub** or **Sign in with GitLab**. Follow the on-screen prompts to grant Semgrep the needed permissions and proceed.
```
Do you remember which of those permissions they asked for? We have a list of permissions, but I believe they are required at different stages of the user journey now.

Reviewer: Can we add a section somewhere (maybe at the bottom) for users who may want to know this prior to signing up? You can reuse from this: [https://semgrep.dev/docs/semgrep-cloud-platform/getting-started/#requested-permissions-for-github-and-gitlab](https://semgrep.dev/docs/semgrep-cloud-platform/getting-started/#requested-permissions-for-github-and-gitlab)

2. Provide the **Organization display name** you'd like to use, then click **Create new organization**.

> 3. Launch your CLI, and follow the instructions on the **Scan a project on your machine** page. For your convenience, we've reproduced that information here and added instructions for Windows users.

Reviewer: ```suggestion
3. Launch your CLI, and follow the instructions on the **Scan a project on your machine** page. The following section reproduces that information and adds instructions for Windows users.
```
- Is the "Scan a project on your machine" page supposed to be a link?
- From the [google style guide:](https://developers.google.com/style/word-list#we) "...It's OK to use we to refer to the organization that's represented as the author of the document as long as the antecedent is clear." - we'd have to identify who "we" are in a document, so we writers do it sparingly.

Whole bit about it: [Second and first person](https://developers.google.com/style/person)

> 3. Launch your CLI, and follow the instructions on the [**Scan a project on your machine**](https://semgrep.dev/onboarding/scan) page. For your convenience, the same information is presented below, along with instructions for Windows users.

Reviewer: Just curious - couldn't we have started from the CLI? If you type in `semgrep login`, it would have opened the login page without the user having to navigate to it, and prompted the user to create an org, and the user would have "expected" to go back to the CLI at some point. Right now, we start with the online app, then move them to the CLI. Is this what happened with the onboarding tour, too? Seems weird that we bring them out of the app out of nowhere.

Reviewer: I am resolving as this is as intended by the product folks. Thank you for bearing with me!

    <Tabs
        defaultValue="macOS"
        values={[
        {label: 'macOS', value: 'macOS'},
        {label: 'Linux', value: 'Linux'},
        {label: 'Windows Subsystem for Linux (WSL)', value: 'Windows Subsystem for Linux (WSL)'},
        {label: 'Docker', value: 'Docker'},
        ]}
    >

    <TabItem value='macOS'>

>     1. Install the Semgrep CLI and confirm the installation:

Reviewer: (Optional / no need to take any action) I can't help but notice that the spacing for the instructions seems looser than in other pages - try checking this out: [http://localhost:3000/docs/running-rules/#yaml-defined-rules][(url](http://localhost:3000/docs/running-rules/#yaml-defined-rules)) and use the inspect tool - it looks like yours is adding a p after the li, [which is totally valid html](https://stackoverflow.com/questions/7660575/is-p-semantically-correct-within-an-li-element). If it bothers you that the other lists are not <p>'ed and have tighter spacing, then you can probably remove the newline before each step, but if it looks ok to you, just leave it as is :) I'm fine with not doing any changes to the page.

        ```console
        # with homebrew
        brew install semgrep

        # with pip
        python3 -m pip install semgrep

        # confirm
        semgrep --version
        ```

>     2. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

Reviewer: Just confirming - Is this copy pasted from the app?

Author: It isn't -- the UI is much more terse:

![image](https://github.com/semgrep/semgrep-docs/assets/16521651/85f37097-642c-43e5-9e87-100d07e8e422)

Reviewer: ```suggestion
    2. Log in to your Semgrep account. Running this command launches a browser window, but you can also use the link that's returned in the CLI to proceed):
```

        ```console
        semgrep login
        ```

    3. In the **Semgrep CLI login**, click **Activate** to proceed.

>     4. Return to the CLI, navigate to the root of your repo, and run your first scan:

Reviewer: ```suggestion
    4. Return to the CLI, navigate to the root of your repository, and run your first scan:
```
See the word list when in doubt: [https://developers.google.com/style/word-list?hl=en#repo](https://developers.google.com/style/word-list?hl=en#repo)

        ```console
        semgrep ci
        ```

    </TabItem>

    <TabItem value='Linux'>

    1. Install the Semgrep CLI and confirm the installation:

        ```console
        # install
        python3 -m pip install semgrep

        # confirm
        semgrep --version
        ```

>     2. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

Reviewer: ```suggestion
    2. Log in to your Semgrep account. Running this command launches a browser window, but you can also use the link that's returned in the CLI to proceed):
```

I just thought of this now - will the link from the CLI include the token?

Author: Yes, it does: `https://semgrep.dev/login?cli-token=<token>&docker=False&gha=False`

        ```console
        semgrep login
        ```

    3. In the **Semgrep CLI login**, click **Activate** to proceed.

>     4. Return to the CLI, navigate to the root of your repo, and run your first scan:

Reviewer: ```suggestion
    4. Return to the CLI, navigate to the root of your repository, and run your first scan:
```

        ```console
        semgrep ci
        ```

    </TabItem>

    <TabItem value='Windows Subsystem for Linux (WSL)'>

    :::info Prerequisites
    Ensure that you have the [Windows Subsystem for Linux installed](https://learn.microsoft.com/en-us/windows/wsl/install) before proceeding.
    :::

    1. Using the WSL interface, install the Semgrep CLI and confirm the installation:

        ```console
        #install
        python3 -m pip install semgrep

        # confirm
        semgrep --version
        ```

>     2. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

Reviewer: (See previous notes)

        ```console
        semgrep login
        ```
    
    3. In the **Semgrep CLI login**, click **Activate** to proceed.
    

>     4. Return to the WSL interface, navigate to the root of your repo, and run your first scan:

Reviewer: ```suggestion
    4. Return to the WSL interface, navigate to the root of your repository, and run your first scan:
```

        ```console
        semgrep ci
        ```

    </TabItem>

    <TabItem value='Docker'>

    :::info Prerequisites

>     Ensure that you have the [Docker installed](https://docs.docker.com/desktop/) before proceeding.

Reviewer: ```suggestion
    Ensure that you have [Docker installed](https://docs.docker.com/desktop/) before proceeding.
```
Yet another proper noun, so we don't use the. Neither does Docker for their docs. [https://docs.docker.com/engine/install/](https://docs.docker.com/engine/install/)

    :::

    1. Pull the latest image and confirm the version:

        ```console
        docker pull returntocorp/semgrep

        # confirm version
        docker run --rm returntocorp/semgrep semgrep --version
        ```

    2. For users running Docker on **macOS or Linux** Docker:

>         1. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

Reviewer: ```suggestion
        1. Log in to your Semgrep account. Running this command launches a browser window, but you can also use the link that's returned in the CLI to proceed:
```

            ```console
            docker run -it returntocorp/semgrep semgrep login
            ```

        2. In the **Semgrep CLI login**, click **Activate** to proceed. Return to the CLI and copy the login token that's shown.

>         3. Navigate into the root of your repo, and run your first scan (be sure to substitute <code><span class="placeholder">YOUR_TOKEN</span></code> with the login token value you copied in the previous step): 

Reviewer: ```suggestion
        3. Navigate into the root of your repository, and run your first scan. Be sure to substitute <code><span class="placeholder">YOUR_TOKEN</span></code> with the login token value you copied in the previous step: 
```

            ```console
            docker run -e SEMGREP_APP_TOKEN=YOUR_TOKEN --rm -v "${PWD}:/src" returntocorp/semgrep semgrep ci
            ```

>             The provided `-v` option mounts the current directory into the container to be scanned. Navigate into a different repo or provide a specific local directory in the command to scan a different project.

Reviewer: ```suggestion
            The provided `-v` option mounts the current directory into the container to be scanned. Navigate into a different repository or provide a specific local directory in the command to scan a different project.
```

    
    3. For users running Docker on **Windows**:

        1. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

            ```console
            docker run -it returntocorp/semgrep semgrep login
            ```

        2. In the **Semgrep CLI login**, click **Activate** to proceed. Return to the CLI, and copy the login token that's shown.

        3. Navigate into the root of your repo, and run your first scan (be sure to substitute <code><span class="placeholder">YOUR_TOKEN</span></code> with the login token value you copied in the previous step): 

            ```console
            docker run -e SEMGREP_APP_TOKEN=YOUR_TOKEN --rm -v "%cd%:/src" returntocorp/semgrep semgrep ci
            ```

>             The provided `-v` option mounts the current directory into the container to be scanned. Navigate into a different repo or provide a specific local directory in the command to scan a different project.

Reviewer: ```suggestion
            The provided `-v` option mounts the current directory into the container to be scanned. Navigate into a different repository or provide a specific local directory in the command to scan a different project.
```

    </TabItem>

    </Tabs>

> 4. Once you've scanned your first application, return to the Cloud Platform, and click **View findings** to see the security vulnerabilities in your project. Alternatively, you can view your results in the Cloud Platform's **Dashboard** page. For detailed information, click **Code** to access your SAST findings or **Supply Chain** to access your SCA findings.

Reviewer: Noting this line down pending results of our To Use Full Name or Not To Use Full Name discussion.

> 4. Once you've scanned your first application, return to Semgrep Cloud Platform (SCP), and click **View findings** to see the security vulnerabilities in your project. Alternatively, you can view your results in SCP's **Dashboard** page. For detailed information, click **Code** to access your SAST findings or **Supply Chain** to access your SCA findings.

Reviewer: ```suggestion
4. Once you've scanned your first application, return to Semgrep Cloud Platform, and click **View findings** to see the security vulnerabilities in your project. Alternatively, you can view your results in SCP's **Dashboard** page. For detailed information, click **Code** to access your SAST findings or **Supply Chain** to access your SCA findings.
```
We already introduced the acronym at the start of the doc :)

    <Tabs
        defaultValue="macOS"
        values={[
        {label: 'macOS', value: 'macOS'},
        {label: 'Linux', value: 'Linux'},
        {label: 'Windows Subsystem for Linux (WSL)', value: 'Windows Subsystem for Linux (WSL)'},
        {label: 'Docker', value: 'Docker'},
        ]}
    >

    <TabItem value='macOS'>

    1. Install the Semgrep CLI and confirm the installation:

        ```console
        # with homebrew
        brew install semgrep

        # with pip
        python3 -m pip install semgrep

        # confirm
        semgrep --version
        ```

        :::note
        **Homebrew users:** ensure that you've [added Homebrew to your PATH](https://docs.brew.sh/FAQ#my-mac-apps-dont-find-homebrew-utilities).
        :::

    2. Log in to your Semgrep account. Running this command launches a browser window, but you can also use the link that's returned in the CLI to proceed:

        ```console
        semgrep login
        ```

    3. In the **Semgrep CLI login**, click **Activate** to proceed.

    4. Return to the CLI, navigate to the root of your repository, and run your first scan:

        ```console
        semgrep ci

>         # Alternatively, run an offline-only SAST scans; no SCA scan is performed

Reviewer: ```suggestion
            # Alternatively, run an offline-only SAST scan; no SCA scan is performed
```

I see that you put it here, but it sort of negates the whole flow because the user doesn't have to sign in for this. I apologize - maybe better to revert in keeping in line with product.

        # with this command
        semgrep scan --config=auto
        ```

    </TabItem>

    <TabItem value='Linux'>

    1. Install the Semgrep CLI and confirm the installation:

        ```console
        # install
        python3 -m pip install semgrep

        # confirm
        semgrep --version
        ```

    2. Log in to your Semgrep account. Running this command launches a browser window, but you can also use the link that's returned in the CLI to proceed:

        ```console
        semgrep login
        ```

    3. In the **Semgrep CLI login**, click **Activate** to proceed.

    4. Return to the CLI, navigate to the root of your repository, and run your first scan:

        ```console
        semgrep ci

>         # Alternatively, run an offline-only SAST scans; no SCA scan is performed

Reviewer: 🙏 (See other comments please).

    <TabItem value='Docker'>

    :::info Prerequisites
    Ensure that you have [Docker installed](https://docs.docker.com/desktop/) before proceeding.
    :::

    1. Pull the latest image and confirm the version:

        ```console
        docker pull returntocorp/semgrep

        # confirm version
        docker run --rm returntocorp/semgrep semgrep --version
        ```

    2. For users running Docker on **macOS or Linux** Docker:

        1. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

            ```console
            docker run -it returntocorp/semgrep semgrep login
            ```

        2. In the **Semgrep CLI login**, click **Activate** to proceed. Return to the CLI and copy the login token that's shown.

        3. Navigate into the root of your repository, and run your first scan. Be sure to substitute <code><span class="placeholder">YOUR_TOKEN</span></code> with the login token value you copied in the previous step: 

            ```console
            docker run -e SEMGREP_APP_TOKEN=YOUR_TOKEN --rm -v "${PWD}:/src" returntocorp/semgrep semgrep ci

>             # Alternatively, run an offline-only SAST scans; no SCA scan is performed

Reviewer: Suggest removing this for now.

            docker run -e SEMGREP_APP_TOKEN=YOUR_TOKEN --rm -v "${PWD}:/src" returntocorp/semgrep semgrep scan --config=auto
            ```

            The provided `-v` option mounts the current directory into the container to be scanned. Navigate into a different repository or provide a specific local directory in the command to scan a different project.
    
    3. For users running Docker on **Windows**:

        1. Log in to your Semgrep account (running this command will launch a browser window, but you can also use the link that's returned in the CLI to proceed):

            ```console
            docker run -it returntocorp/semgrep semgrep login
            ```

        2. In the **Semgrep CLI login**, click **Activate** to proceed. Return to the CLI, and copy the login token that's shown.

        3. Navigate into the root of your repository, and run your first scan. Be sure to substitute <code><span class="placeholder">YOUR_TOKEN</span></code> with the login token value you copied in the previous step: 

            ```console
            docker run -e SEMGREP_APP_TOKEN=YOUR_TOKEN --rm -v "%cd%:/src" returntocorp/semgrep semgrep ci
            

>             # Alternatively, run an offline-only SAST scans; no SCA scan is performed

Reviewer: ```suggestion
            # Alternatively, run an offline-only SAST scan; no SCA scan is performed
```

I see that you put it here, but it sort of negates the whole flow because the user doesn't have to sign in for this. I apologize - maybe better to revert in keeping in line with product.
